### ISSUE TITLE
Accessibility improvements on Button component

### ISSUE THREAD

**Issue by muratcorlu** [Trendyol/baklava#285]
- [x] Buttons should be focusable with keyboard navigation.
- [x] Focus state should be visible for keyboard navigation.
- [x] `Space` and `Enter` keys should trigger button action while it's focused.
- [ ] `type="submit"` should submit it's wrapping form

**Comment by AykutSarac** [comment]
Hello, are there any ETA on it?

**Comment by muratcorlu** [comment]
We are planning to pick some of the a11y tasks next week. But PRs are always welcome 😊
